src/metabase/api/card.clj

```diff
@@ -365,6 +365,14 @@ saved later when it is ready."
       (when timed-out?
         (schedule-metadata-saving result-metadata-chan <>)))))
 
+(defn- check-valid-card-info
+  "Checks that the card is valid before saving. Throws an exception if not."
+  [{:keys [dataset dataset_query]}]
+  (let [template-tag-types (->> (vals (get-in dataset_query [:native :template-tags]))
+                                (map (comp keyword :type)))]
+    (when (and dataset (some (complement #{:card :snippet}) template-tag-types))
+      (throw (ex-info (tru "A model made from a native SQL question cannot have a variable or field filter.") {:status-code 400})))))
```

Comment:
nit: I think it looks cleaner if have `{:status-code 400}`  in a new line.

Reply:
Fixed
https://github.com/metabase/metabase/blob/5be754da9da32b5794af5554b739a21ae31f1894/src/metabase/models/card.clj#L211